### PR TITLE
feat: add chart companion components

### PR DESCRIPTION
**Pull request by kristw**
🏆 Enhancements

* Create new package `@superset-ui/chart-companion` which includes the small things that are used to composite a chart. 
* Code are migrated from `@superset-ui/plugins` 's `preset-chart-xy` 
* Add unit tests for 100% coverage

### CODE REVIEW

File: packages/superset-ui-chart-companion/src/legend/WithLegend.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    const { className, width, height, position, renderChart, renderLegend } = this.props;
+
+    const isHorizontal = position === 'left' || position === 'right';
+
+    const style: CSSProperties = {
+      display: 'flex',
+      flexDirection: this.getContainerDirection(),
+    };
+
+    style.width = width;
```
Comment (williaster) [inline packages/superset-ui-chart-companion/src/legend/WithLegend.tsx]: why not just set these when you define `style`? `const style = { ..., width, height };`

File: packages/superset-ui-chart-companion/src/legend/WithLegend.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
+/* eslint-disable sort-keys */
+import React, { CSSProperties, ReactNode, PureComponent } from 'react';
+import { ParentSize } from '@vx/responsive';
+// eslint-disable-next-line import/no-unresolved
+import * as CSS from 'csstype';
```
Comment (williaster) [inline packages/superset-ui-chart-companion/src/legend/WithLegend.tsx]: can you just import `FlexDirectionProperty`?

File: packages/superset-ui-chart-companion/test/legend/WithLegend.test.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      done();
+    }, 500);
+  });
+
+  it('renders without width or height', done => {
+    const wrapper = mount(<WithLegend renderChart={renderChart} renderLegend={renderLegend} />);
+
+    triggerResizeObserver();
+    // Have to delay > 300ms
+    // because the default debounce time for ParentSize is 300.
```
Comment (williaster) [inline packages/superset-ui-chart-companion/test/legend/WithLegend.test.tsx]: you could expose this prop so this is more declarative.

**Comment by kristw** [comment]
`chart-composition` sounds good

File: packages/superset-ui-chart-companion/test/legend/WithLegend.test.tsx (hunk 3 above)
Comment (kristw) [reply packages/superset-ui-chart-companion/test/legend/WithLegend.test.tsx]: great comment!

File: packages/superset-ui-chart-companion/src/legend/WithLegend.tsx (hunk 1 above)
Comment (kristw) [reply packages/superset-ui-chart-companion/src/legend/WithLegend.tsx]: ah true. delete the `if` condition that is no longer needed and forgot to look around.
will fix.

**Pull request merged**